JsonPointer/JsonPointer.cs

```diff
@@ -253,6 +253,17 @@ public static JsonPointer Create<T>(Expression<Func<T, object>> expression)
 				{
 					segments.Insert(0, PointerSegment.Create(mce.Arguments[0].ToString()));
 					body = mce.Object;
+				} else if (body is BinaryExpression
+				           {
+					           Right: ConstantExpression arrayIndexConstantExpression
+				           } binaryExpression and
+				           {
+					           NodeType: ExpressionType.ArrayIndex
+				           })
```

Comment:
This is so weird.  I don't like formatting `if`s on multiple lines unless each line is a sequence of conditionals, but I'm not sure how else do this nicely.

Reply:
I would 
(a) be less strict about having the brackets on multiple lines (most examples of pattern matching don't do this anyways), and 
(b) put each `and` on its own line. 

Like 
```csharp
body is BinaryExpression { Right: ConstantExpression arrayIndexConstantExpression }
    and { NodeType: ExpressionType.ArrayIndex  }
```

Reply:
There's actually a variable declaration in there, too, so it'd be

```c#
body is BinaryExpression { Right: ConstantExpression arrayIndexConstantExpression } binaryExpression
    and { NodeType: ExpressionType.ArrayIndex  }
```

I tend to like `&&` at the end of lines, but `and` at the end is too long. 🤷‍♂️

Reply:
Yeah, I think that's the best I'm going to do.

Reply:
I know you completed this PR already, but there's this alternative for a future PR:
```csharp
body is BinaryExpression {
    Right: ConstantExpression arrayIndexConstantExpression,
    NodeType: ExpressionType.ArrayIndex 
} binaryExpression
```